### PR TITLE
Add include/exclude regexes for assetstore import process

### PR DESCRIPTION
**Pull request by zachmullen**

### CODE REVIEW

File: girder/api/v1/assetstore.py [hunk 1]
```diff
@@ -150,6 +150,10 @@ def createAssetstore(self, params):
                enum=('folder', 'collection', 'user'))
         .param('progress', 'Whether to record progress on the import.',
                dataType='boolean', default=False, required=False)
+        .param('fileIncludeRegex', 'If set, only filenames matching this regular '
+               'expression will be imported.', required=False)
+        .param('fileExcludeRegex', 'If set, only filenames that do not match this regular '
+               'expression will be imported.', required=False)
```
Comment (jeffbaumes) [inline girder/api/v1/assetstore.py]: May be good to clarify with "If a filename matches both inclusion and exclusion patterns, exclusion takes precedence over inclusion." in the docs here.

**Comment by zachmullen** [comment]
@manthey PTAL, I noticed we needed this feature in our histomicsTK provisioning scripts in order to exclude the .gitignore file from the import.

File: tests/cases/assetstore_test.py [hunk 2]
```diff
@@ -193,12 +188,26 @@ def testFilesystemAssetstoreImport(self):
...
+        # Test importing directory with include & exclude regexes; file should be excluded
+        params['importPath'] = os.path.join(ROOT_DIR, 'tests', 'cases', 'py_client')
+        self.request(
+            path, method='POST', params=dict(
+                params, fileIncludeRegex='world.*', fileExcludeRegex='world.*'))
+        resp = self.request('/resource/lookup', user=self.admin, params={
+            'path': '/user/admin/Public/world.txt/world.txt',
+            'test': True
+        })
+        self.assertStatusOk(resp)
```
Comment (manthey) [inline tests/cases/assetstore_test.py]: The coverage report implies that this call isn't doing what we expect (the line where the exclusion filter occurs is reported as uncovered).

File: tests/cases/assetstore_test.py [hunk 3]
```diff
@@ -193,12 +188,26 @@ def testFilesystemAssetstoreImport(self):
...
-        params['importPath'] = os.path.join(
-            ROOT_DIR, 'tests', 'cases', 'py_client')
-        resp = self.request(path, method='POST', params=params, user=self.admin)
+        self.model('file').remove(file)
+
+        # Test importing directory with include & exclude regexes; file should be excluded
+        params['importPath'] = os.path.join(ROOT_DIR, 'tests', 'cases', 'py_client')
+        self.request(
+            path, method='POST', params=dict(
+                params, fileIncludeRegex='world.*', fileExcludeRegex='world.*'))
```
Comment (manthey) [inline tests/cases/assetstore_test.py]: This is missing `user=self.admin` as a parameter to the POST call.

Comment (zachmullen) [inline tests/cases/assetstore_test.py]: Good catch, I forgot to assert that this request returned OK.

**Comment by manthey** [comment]
LGTM

**Pull request merged**